torch/_inductor/kernel/mm_common.py

```
@@ -10,13 +10,29 @@
 
 from .. import config as inductor_config
 from ..codegen.wrapper import PythonWrapperCodegen
-from ..ir import Layout
-from ..utils import ceildiv as cdiv, get_num_sms, TMA_DESCRIPTOR_SIZE
+from ..ir import ChoiceCaller, Layout
+from ..utils import (
+    ceildiv as cdiv,
+    get_num_sms,
+    TMA_DESCRIPTOR_SIZE,
+    use_aten_gemm_kernels,
+)
 
 
 log = logging.getLogger(__name__)
 
 
+def should_fallback_to_aten(choices: list[ChoiceCaller]) -> bool:
+    fallback_to_aten: bool = (
+        len(choices) == 0
+        and not use_aten_gemm_kernels()
+        and inductor_config.autotune_fallback_to_aten
```

Do we need `autotune_fallback_to_aten` when `max_autotune_gemm_backends` exists ? isnt this duplicative ?

---

> Do we need `autotune_fallback_to_aten` when `max_autotune_gemm_backends` exists ? isnt this duplicative ?

I think the intent is to be super safe. Even if users specify only "CUTLASS" and CUTLASS fails, it will still keep things running. I think this safe logic predates the autotune_fallback_to_aten config. 

But yeah it has been pretty painful for me when working on cutlass since bunch of stuff are failing silently.

---

If the user wants to fallback, they should do `max_autotune_gemm_backends="CUTLASS, ATEN"`. I dont think we should have two ways of doing the same exact thing.

---

@eellison I agree with the idea. I can commit to removing autotune_fallback_to_aten all together and removing the silent fallback logic. But it will take a few PRs and a while. Does that sound good?